mobile-app/app/screens/AppNavigator/screens/Balances/hooks/DisplayAnnouncement.ts

```diff
@@ -0,0 +1,46 @@
+import { DisplayAnnouncementPersistence } from '@api/persistence/display_announcement_storage'
+import { useLogger } from '@shared-contexts/NativeLoggingProvider'
+import { useEffect, useState } from 'react'
+
+export interface DisplayAnnouncement {
+  hiddenAnnouncements: string[]
+  hideAnnouncement: (id: string) => void
+}
+
+export interface AnnouncementFlag {
+  id: string
+  displayAnnouncement: boolean
+}
+
+export function useDisplayAnnouncement (): DisplayAnnouncement {
+  const logger = useLogger()
+  const [hiddenAnnouncements, setHiddenAnnouncements] = useState<string[]>([])
+
+  useEffect(() => {
+    let mounted = true
+    DisplayAnnouncementPersistence.get().then((flag: string[]) => {
```

Comment:
there's something weird with the format here

Reply:
this is the cleanup function that i am adding to prevent the memory leak if component unmounts when the `DisplayAnnouncementPersistence` is still retrieving

Reply:
No, i mean there's a weird syntax format. The brackets are misaligned